### ISSUE TITLE
Manager seems to pick the first queue in the list

### ISSUE THREAD

**Issue by jmhirsch** [swar/Swar-Chia-Plot-Manager#898]
When creating new plots, if multiple queues are past their minimum stagger time, the manager seems to pick the first one in the list.
 
This results in drives near the top in the list being picked more often, and drives near the bottom of the list often running less than the maximum possible number of plots.

Overtime, plotting output slows down as the drives at the top are running at capacity while the drives at the bottom are underutilized. 

A possible solution would be to introduce randomness when picking the next drive, or to pick the drive where a plot has not been kicked off for the longest amount of time. 

To clarify, this is a problem when using a single Temp directory in each queue. 

This issue seems consistent for me across two machines with different configurations

**Comment by Tuckie** [comment]
I would love a mode that just always picks the drive with the most space on it; whether it is the temporary directory or the final.

**Comment by flynn-buc** [comment]
> I would love a mode that just always picks the drive with the most space on it; whether it is the temporary directory or the final.

I like that idea a lot, but when plotting, most space used isn't always accurate. I can have a 2TB drive with 2 plots and a 1TB drive with 0 plots, and the better solution would be to pick the 1TB drive, but going by available capacity returns the 2TB drive. Going by used % could resolve this issue, but I've seen multiple cases where the % was incorrect, so I'm not sure if I would want to rely on that. 

For destination drives, your idea works really well though (provided you don't copy more than 1 file to 1 drive at a time)

**Comment by Tuckie** [comment]
Yeah, estimating the plotting side of things definitely isn't perfect, but it would be good enough to allow for a single job with a variety of temp disk speeds. Today if you try to do that it will just fill up the slow drive with jobs.

Another way to frame this as we look to optimize the system:

How best to balance constraints:
 1. Temp drive space vs iops/io wait - Generally speaking, it is best to pick the drive with the most space, however, to your point, what we're really looking for is the drive that has at least enough space with the least iowait. This can be simplified to a current jobs/max jobs (user specified) percentage for each temp drive and the system can just pick the drive with the lowest activity percentage. This could also make plotting on spinning rust easier as one can just set the max jobs to 1 for a drive.
 
 2. Destination drive space vs iops/io wait - Honestly, I would probably tackle this the same way as above, that way we can get back to plotting ASAP, but there's definitely an argument to be made for just pushing it to the disk with the most space (either percentage or GB) 
 
 3. CPU cores - This one is mainly why I think the jobs system needs refinement into a single queue. With temp drives or destination drives operating at different speeds, there is no reliable way to cascade usage in such a way where we can give as many threads as possible to that first step without random job waits as the start timing shifts between queues.

**Comment by gabrialm** [comment]
> For destination drives, your idea works really well though (provided you don't copy more than 1 file to 1 drive at a time)

Easy way to solve for this is to utilize something like MergerFS (like I am in Linux), or DrivePool in Windows, combined with a "most free space" policy, this will ensure that files copied to your pool target will be even displaced across all drives in the pool; that assumes of course that all of your drives are uniform in capacity across the pool, and that your pool is dedicated to nothing by Chia plots.

**Comment by Tuckie** [comment]
> Easy way to solve for this is to utilize something like MergerFS

I've definitely been toying with that idea, I just fear that adding that layer would cause a performance impact.
